packages/react/src/components/Slider/Slider.test.tsx

```diff
@@ -34,95 +25,94 @@ describe('Slider', () => {
+    const { getByRole } = render(<Slider aria-labelledby="custom-label" />);
+    expect(getByRole('slider').getAttribute('aria-labelledby')).toBe('custom-label');
   });
 
   it('can provide the current value', () => {
     const slider = React.createRef<ISlider>();
 
-    wrapper = mount(<Slider defaultValue={12} min={0} max={100} componentRef={slider} />);
+    render(<Slider defaultValue={12} min={0} max={100} componentRef={slider} />);
     expect(slider.current?.value).toEqual(12);
   });
 
   it('can provide the current range', () => {
     const slider = React.createRef<ISlider>();
 
-    wrapper = mount(<Slider defaultValue={12} min={0} max={100} componentRef={slider} ranged />);
+    render(<Slider defaultValue={12} min={0} max={100} componentRef={slider} ranged />);
     expect(slider.current?.range).toEqual([0, 12]);
   });
 
   it('can set id', () => {
-    wrapper = mount(<Slider id="test_id" styles={{ titleLabel: 'test_label' }} />);
+    const { container, getByRole } = render(<Slider id="test_id" styles={{ titleLabel: 'test_label' }} />);
 
-    const sliderSlideBox = wrapper.find('.ms-Slider-slideBox');
-    expect(sliderSlideBox.getDOMNode().id).toEqual('test_id');
+    expect(getByRole('slider').id).toEqual('test_id');
 
     // properly associates label with custom id
-    const label = wrapper.find('label.test_label');
-    expect(label.prop('htmlFor')).toBe('test_id');
+    const label = container.getElementsByClassName('test_label')[0];
+    expect(label.getAttribute('for')).toBe('test_id');
   });
 
   it('can set id via buttonProps', () => {
     // Not the recommended way of doing things, but it should work consistently still
-    wrapper = mount(<Slider buttonProps={{ id: 'test_id' }} styles={{ titleLabel: 'test_label' }} />);
+    const { container, getByRole } = render(
+      <Slider buttonProps={{ id: 'test_id' }} styles={{ titleLabel: 'test_label' }} />,
+    );
 
-    const sliderSlideBox = wrapper.find('.ms-Slider-slideBox');
-    expect(sliderSlideBox.getDOMNode().id).toEqual('test_id');
+    expect(getByRole('slider').id).toEqual('test_id');
 
     // properly associates label with custom id
-    const label = wrapper.find('label.test_label');
-    expect(label.prop('htmlFor')).toBe('test_id');
+    const label = container.getElementsByClassName('test_label')[0];
+    expect(label.getAttribute('for')).toBe('test_id');
   });
 
   it('handles zero default value', () => {
     const slider = React.createRef<ISlider>();
 
-    wrapper = mount(<Slider defaultValue={0} min={-100} max={100} componentRef={slider} />);
+    render(<Slider defaultValue={0} min={-100} max={100} componentRef={slider} />);
     expect(slider.current!.value).toEqual(0);
   });
 
   it('handles zero value', () => {
     const slider = React.createRef<ISlider>();
 
-    wrapper = mount(<Slider value={0} min={-100} max={100} componentRef={slider} />);
+    render(<Slider value={0} min={-100} max={100} componentRef={slider} />);
```

Comment:
Not a new issue but it would be nice if we could test something in the DOM to make sure the correct value is rendered as well. Since this implementation doesn't use a native input, you'd have to look if there are any aria attributes or something that you could check.

Reply:
If I'm understanding this correctly, I think this should be covered by the `renders correct aria-valuetext` test below

Reply:
Right, I guess it just concerns me somewhat that we're checking the internal component state without checking what it's actually rendering (in case there's a mismatch)